lib/shape.mli

```diff
@@ -6,6 +6,7 @@ val circle : ?x:int -> ?y:int -> int -> shape
 val rectangle : ?x:int -> ?y:int -> int -> int -> shape
 val ellipse : ?x:int -> ?y:int -> int -> int -> shape
 val show : shape list -> unit
+val scale: int -> shape -> shape
```

Comment:
Hmm, on second thoughts, we might want this to be `float -> shape -> shape` since something like `scale 0.5 c` is a plausible usage of this transformation. What do you think @nikochiko @AryanGodara?

Reply:
Awesome. Both of these suggestions make sense. I will complete this soon. Thanks

Reply:
I have a question regarding this. Since the length/radius of a shape can only be an integer. Do I round the radius to the nearest integer or will there be some changes in the definition of the shape?

Reply:
Also about scaling, in python implementation, I found that the scaling factor for both the x direction and y direction is taken as parameters. Is this how we go as well?

Reply:
> Since the length/radius of a shape can only be an integer. Do I round the radius to the nearest integer or will there be some changes in the definition of the shape?

Rounding off to the nearest integer sounds reasonable to me.

> I found that the scaling factor for both the x direction and y direction is taken as parameters. Is this how we go as well?

Good point. I think one way might be to use poymorphic variants to specify what scaling one wants. Here's an example of their usage -

```ocaml
let find_scale s =
match s with
| `Scale_X x -> Printf.printf "x = %d\n" x
| `Scale_Y y -> Printf.printf "y = %d\n" y
| `Scale (x, y) -> Printf.printf "x = %d, y = %d" x y;;
```
The idea is that, if you have just one parameter specified it applies to both x and y axes. This would also mean we should use the variant as an argument for the scaling factor. I'm open to suggestions on this one.